e2e/testcases/helm_sync_test.go

```diff
@@ -935,14 +942,27 @@ func fetchServiceAccountKeyFile(nt *nomostest.NT, projectID, gsaKeySecretID, gsa
 		"--format", "value(name)",
 		"--project", projectID)
 	if err != nil {
-		// TODO: create new key and secret if no secret found
-		return "", fmt.Errorf("listing enabled secret versions: %w", err)
+		if !strings.Contains(string(out), "NOT_FOUND") {
+			return "", err
+		}
+		nt.T.Log("Secret not found, bootstrapping initial Secret")
+		_, err = nt.Shell.ExecWithDebug("gcloud", "secrets", "create",
+			gsaKeySecretID,
+			"--project", projectID)
+		if err != nil {
+			return "", fmt.Errorf("bootstrapping secret: %w", err)
+		}
+		err = generateServiceAccountKey(nt, projectID, gsaKeySecretID, gsaEmail, gsaKeyFilePath)
+		if err != nil {
+			return "", err
+		}
+		return gsaKeyFilePath, nil
 	}
 	// For some reason, the "name" value is just the version,
 	// even tho in json the "name" field is the full name.
 	// But what the access command wants is the version. So it's fine.
 	gsaKeySecretVersion := strings.TrimSpace(string(out))
-	if gsaKeySecretVersion == "" {
+	if gsaKeySecretVersion == "" || strings.Contains(gsaKeySecretVersion, "WARNING") {
```

Comment:
what's the warning edge case?

Reply:
If you list a secret that exists but doesn't have any versions, it prints a warning message but doesn't return an error

Reply:
Another approach could be trying to parse the output to an integer and failing if it errors

Reply:
Hmm how about matching on lines that start with WARNING and removing the whole line, just in case there's some other warning we don't know about. 

Or check for exact match of the empty warning and error if there are other unexpected warnings.

Reply:
I don't want this to be too brittle, as we're just checking a warning string on the output for errors. If the warning message changes slightly, I'd prefer this not break. 

I updated this to pass in the `--no-user-output-enabled` flag, which suppresses the warning message. This means that when there are no versions the output should more reliably pass the empty check on this line.

Reply:
great, thanks.